### PR TITLE
Add drivermodule plugin and wrap common access to a Drver's names in …

### PR DESCRIPTION
**Pull request by atcuno**
…a method

### CODE REVIEW

**Comment by atcuno** [comment]
Output against a memory sample infected with Ghost Emperor:

```
$ python3 vol.py -r pretty -f ../Gho* drivermodule
Volatility 3 Framework 2.4.0
  |         Offset | Known Exception |       Driver Name |               Serivce Key |          Alternative Name
* | 0xe38a8f9ded20 |            True |               RAW |                           |           \FileSystem\RAW
* | 0xe38a94f022e0 |           False | dump_audio_codec0 | \Driver\dump_audio_codec0 | \Driver\dump_audio_codec0
```

File: volatility3/framework/plugins/windows/drivermodule.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+                yield (0, (format_hints.Hex(driver.vol.offset), known_exception, driver_name, service_key, name))
+
+    def run(self):
+
+        return renderers.TreeGrid([
+            ("Offset", format_hints.Hex),
+            ("Known Exception", bool),
+            ("Driver Name", str),
+            ("Serivce Key", str),
```
Comment (digitalisx) [inline volatility3/framework/plugins/windows/drivermodule.py]: Found typo

```suggestion
            ("Service Key", str),
```

File: volatility3/framework/plugins/windows/drivermodule.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                "PnpManager",
+                "RAW",
+                "WMIxWDM",
+                "Win32k",
+                "Fs_Rec"]
+
+class DriverModule(interfaces.plugins.PluginInterface):
+    """Determines if any loaded drivers were hidden by a rootkit"""
+
+    _required_framework_version = (2, 0, 0)
```
Comment (digitalisx) [inline volatility3/framework/plugins/windows/drivermodule.py:21]: How about adding a version of the plugin itself for maintenance?

```suggestion
    _required_framework_version = (2, 0, 0)
    _version = (1, 0, 0)
```

**Review by digitalisx — COMMENTED** [review]
Thank you for your work @atcuno, That's awesome plugin :)
I leave some comment about this PR.

File: volatility3/framework/plugins/windows/drivermodule.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    @classmethod
+    def get_requirements(cls):
```
Comment (digitalisx) [inline volatility3/framework/plugins/windows/drivermodule.py]:
```suggestion
    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
```

File: volatility3/framework/plugins/windows/drivermodule.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        return [
+            requirements.ModuleRequirement(name = 'kernel', description = 'Windows kernel',
+                                           architectures = ["Intel32", "Intel64"]),
+            requirements.PluginRequirement(name = 'ssdt', plugin = ssdt.SSDT, version = (1, 0, 0)),
+            requirements.PluginRequirement(name = 'driverscan', plugin = driverscan.DriverScan, version = (1, 0, 0)),
+        ]
+
+    def _generator(self):
```
Comment (digitalisx) [inline volatility3/framework/plugins/windows/drivermodule.py]:
```suggestion
    def _generator(self) -> Iterator[Tuple]:
```

File: volatility3/framework/plugins/windows/drivermodule.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            # we do not care about actual symbol names, we just want to know if the driver points to a known module
+            module_symbols = list(collection.get_module_symbols_by_absolute_location(driver.DriverStart))
+            if not module_symbols:
+                driver_name, service_key, name = driverscan.DriverScan.get_names_for_driver(driver)
+
+                known_exception = driver_name in KNOWN_DRIVERS
+
+                yield (0, (format_hints.Hex(driver.vol.offset), known_exception, driver_name, service_key, name))
+
+    def run(self):
```
Comment (digitalisx) [inline volatility3/framework/plugins/windows/drivermodule.py]:
```suggestion
    def run(self) -> renderers.TreeGrid:
```

**Review by digitalisx — COMMENTED** [review]
And some NIT comments about return type! :)

File: volatility3/framework/plugins/windows/drivermodule.py (hunk 3 above)
Comment (ikelos) [reply volatility3/framework/plugins/windows/drivermodule.py]: I don't think many of these have their definition, but I'm ok putting it in...

**Review by ikelos — APPROVED** [review]
Yep, looks good.  I'm pleased that the plugins are coming out short and clear as to what they do, makes me feel like the framework's taking all the hard work out of it, as it should.  5:)  Lemme know when you want to merge this (or feel free to merge it yourself).  5:)

File: volatility3/framework/plugins/windows/drivermodule.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
```
Comment (ikelos) [inline volatility3/framework/plugins/windows/drivermodule.py]:
```suggestion
from typing import Iterator, List, Tuple
```

**Review by digitalisx — APPROVED** [review]
All right @ikelos, Thank you for work about my review. :)

**Pull request merged**